src/include/dhcp-server/dialogs.rb

```diff
@@ -378,27 +378,27 @@ def SectionTypeChoose
       par_id = @parent_id
       par_type = @parent_type
       while par_id != ""
-        parents = Builtins.add(parents, par_type)
+        parents << par_type
 
         par = DhcpServer.GetEntryParent(par_type, par_id)
         par_type = Ops.get(par, "type", "")
         par_id = Ops.get(par, "id", "")
       end
       possible = ["subnet", "host", "shared-network", "group", "pool", "class"]
-      if Builtins.contains(parents, "class") ||
-          Builtins.contains(parents, "host") ||
-          Builtins.contains(parents, "pool")
+      if ["class", "host", "pool"].include?(parents)
```

Comment:
you mean if this code do same think as before? yes. Logic there is that it check if in parents are leaps, which cannot have anything under itself....ah, parents can be also array, let me test. we in fact need intersect here